server/remote/msgraph/get_schedule_batched.go

```diff
@@ -0,0 +1,104 @@
+package msgraph
+
+import (
+	"strconv"
+
+	"github.com/mattermost/mattermost-plugin-msoffice/server/remote"
+)
+
+type GetScheduleSingleResponse struct {
+	ID      string                      `json:"id"`
+	Status  int                         `json:"status"`
+	Body    *remote.GetScheduleResponse `json:"body"`
+	Headers map[string]string           `json:"headers"`
+}
+
+type GetScheduleBatchResponse struct {
+	Responses []*GetScheduleSingleResponse `json:"responses"`
+}
+
+type GetScheduleRequest struct {
+	// List of emails of users that we want to check
+	Schedules []string `json:"schedules"`
+
+	// Overall start and end of entire search window
+	StartTime *remote.DateTime `json:"startTime"`
+	EndTime   *remote.DateTime `json:"endTime"`
+
+	// Size of each chunk of time we want to check
+	// This can be equal to end - start if we want, or we can get more granular results by making it shorter.
+	// For the graph API: The default is 30 minutes, minimum is 6, maximum is 1440
+	// 15 is currently being used on our end
+	AvailabilityViewInterval int `json:"availabilityViewInterval"`
+}
+
+func (c *client) GetSchedule(remoteUserID string, schedules []string, startTime, endTime *remote.DateTime, availabilityViewInterval int) ([]*remote.ScheduleInformation, error) {
```

Comment:
Even though we are using app-level creds, we still need to provide a user id to the request. The body of the request contains several users' information, but the url needs to have one in it as well.

Our options are
```
POST /me/calendar/getSchedule 
POST /users/{id|userPrincipalName}/calendar/getSchedule
```

`me` doesn't work for the app-level user, and I don't think it has a user id. This endpoint specifically supports app-level users, so I'm not sure why we need to provide a user id in the url, or what the point of the of id is if the users we are searching for are in the body.

https://docs.microsoft.com/en-us/graph/api/calendar-getschedule

Reply:
See my comment about using (and mapping) the BotAccount. We can still use special-case the credentials to invoke graph, but require that a user is specified.